src/sql/workbench/api/browser/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -345,6 +342,11 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 		}
 	}
 
+	async $tryCreateNotebookDocument(options: INotebookShowOptions): Promise<UriComponents> {
+		let input = await this._notebookService.createNotebookInput(options);
+		return input?.resource;
```

Comment:
If this can be undefined then the return type should be `UriComponents | undefined`

When would we ever get undefined back from createNotebookInput anyways? That doesn't seem like a valid scenario

Reply:
Sure, although I think it'd probably be better to just have createNotebookInput throw if it can't actually create a Notebook input